interface/presenter/pokemon_presenter.go

```diff
@@ -0,0 +1,53 @@
+package presenter
+
+import (
+	"encoding/json"
+	"math/rand"
+	"time"
+
+	"github.com/wmaldonadoc/academy-go-q42021/domain/model"
+	"github.com/wmaldonadoc/academy-go-q42021/infrastructure/api"
+	"github.com/wmaldonadoc/academy-go-q42021/interface/vendors"
+
+	"go.uber.org/zap"
+)
+
+type pokemonPresenter struct{}
+
+// PokemonPresenter - Holds an abstraction of presenter methods.
+type PokemonPresenter interface {
+	ResponsePokemon(p *model.Pokemon) *model.Pokemon
+	ResponseMappedPokemonFromAPI(p *api.ApiResponse) *model.Pokemon
+}
+
+// NewPokemonPresenter - Create and returns a concret instance of pokemonPresenter.
+func NewPokemonPresenter() *pokemonPresenter {
```

Comment:
Be careful with the type of access of your returns. pokemonPresenter is private, and returning an unexported type might lead to future refactors.

Reply:
This was not resolved.